internal/controlplane/common.go

```diff
@@ -67,13 +67,45 @@ func getProviderFromRequestOrDefault(
 			len(providers))
 	}
 
+	return findProvider(in.GetContext().GetProvider(), providers)
+}
+
+func listProvidersOrInferDefault(
+	ctx context.Context,
+	store db.Store,
+	in HasProtoContext,
+	projectId uuid.UUID,
+) ([]db.Provider, error) {
+	// Allows us to take into account the hierarchy to find the provider
+	parents, err := store.GetParentProjects(ctx, projectId)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "cannot retrieve parent projects: %s", err)
+	}
+	providers, err := store.ListProvidersByProjectID(ctx, parents)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "cannot retrieve providers: %s", err)
+	}
+	// if we do not have a provider name, check if we can infer it
```

Comment:
I don't think this comment is correct anymore -- we're not trying to infer the provider here, we're returning all matching providers.